packages/models/src/classMap.js

```diff
@@ -1,64 +1,40 @@
 import CodeObject, { CodeObjectType } from './codeObject';
 
-// Recursively add a code object to an array
-// If the code object exists in the array already, add its children to the
-// existing code object instead.
-function addCodeObject(codeObjectArray, codeObject, parent = null) {
-  if (
-    !codeObjectArray ||
-    !Array.isArray(codeObjectArray) ||
-    !codeObject ||
-    !(codeObject instanceof CodeObject)
-  ) {
-    return;
-  }
-
-  // TODO.
-  // This ignores static/non-static function collisions and function overloads, though this method
-  // is never currently called in a context where those edge cases exist.
-  const existingObject = codeObjectArray.find(
-    (obj) => obj.type === codeObject.type && obj.name === codeObject.name
-  );
-
-  if (!existingObject) {
-    codeObject.parent = parent; // eslint-disable-line no-param-reassign
-    codeObjectArray.push(codeObject);
-  } else {
-    codeObject.children.forEach((child) =>
-      addCodeObject(existingObject.children, child, existingObject)
-    );
-  }
+function indexCodeObject(co, codeObjects, codeObjectsById) {
+  codeObjects.push(co);
+  codeObjectsById[co.id] = co;
 }
 
 export default class ClassMap {
   constructor(classMap) {
     this.codeObjectsByLocation = {};
+    this.constructedCodeObjects = {};
```

Comment:
These are the only code objects for which we used `event.toString()` as a key. Other code objects are keyed by location.